### PR TITLE
Add -I flag

### PR DESCRIPTION
**Pull request by bogem**
Close https://github.com/davecheney/httpstat/issues/1

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -67,14 +67,25 @@ var (
...
+		fmt.Fprintf(os.Stderr, "Usage: httpstat [flags] <url>\nFlags:\n")
+		flag.PrintDefaults()
+	}
 	flag.Parse()
 
 	args := flag.Args()
 	if len(args) != 1 {
-		log.Fatalf("usage: %s URL", os.Args[0])
+		flag.Usage()
 	}
```
Comment (moorereason) [inline main.go]: Need `os.Exit(1)`

**Comment by bogem** [comment]
@moorereason yes, thanks. I will add it

File: main.go (hunk 1 above)
Comment (davecheney) [inline main.go]: Thought that flag.Usage()  called os.Exit(2) after printing.

Comment (davecheney) [inline main.go]: It's easy to test either way.

File: main.go [hunk 2]
```diff
@@ -67,14 +67,26 @@ var (
 			return fmt.Sprintf("\x1b[;38;5;%dm%s\x1b[0m", code+232, s)
 		}
 	}
+
+	// Flag variables
+	onlyHeader bool
 )
 
+func init() {
+	flag.BoolVar(&onlyHeader, "head", false, "show only header of response")
```
Comment (davecheney) [inline main.go]: For rough  compatibility with curl, what do you think about calling this flag `-I`?

File: main.go [hunk 3]
```diff
@@ -141,6 +153,11 @@ func main() {
 		fmt.Println(grayscale(14)(k+":"), cyan(strings.Join(v, ",")))
 	}
 
+	// If -head flag set, then skip other output
+	if onlyHeader {
```
Comment (davecheney) [inline main.go]: What about the timings ? Would it make sense to  do this check earlier and skip reading the body (it's ok just to let the connection close when the program exits).

Suggestion:

add this check at line 140, "if !onlyHeader { ... read body }`

and switch the `resp.Body.Close` to a defer

File: main.go (hunk 2 above)
Comment (bogem) [inline main.go]: It's the same flag:
` -I, --head          Show document info only`
But I can name it `-I`

**Comment by davecheney** [comment]
The standard flag package only supports single hypen arguments even thought
the names of the flags can be longer than a single charater.

I'd prefer this one was called -I. If you want to look at a replacement
flag parser that supports BSD and GNU long arguments, that would be fine,
in another PR.

**Comment by bogem** [comment]
@davecheney ok, I will use single character flag.
Yes, it would be great to use GNU arguments. I'll open a discussion about that in another PR later this day

**Comment by bogem** [comment]
🙏
All conflicts are solved

File: main.go [hunk 4]
```diff
@@ -168,6 +174,11 @@ func visit(url *url.URL) {
 		fmt.Println(grayscale(14)(k+":"), color.CyanString(strings.Join(v, ",")))
 	}
 
+	// if only header is requested, then skip other output
```
Comment (davecheney) [inline main.go]: i don't want you to return at this point, it should continue to read the response but _not_ count the time read the response. That is, -I sets the method to `HEAD` and expects there to be no body.

Comment (bogem) [inline main.go]: Ok. Just updated PR according to your offer

**Pull request merged**